src/views/Ifos/hooks/v3/useGetPublicIfoData.ts

```diff
@@ -0,0 +1,215 @@
+      totalAmountPool: BIG_ZERO,
+      sumTaxesOverflow: BIG_ZERO,
+      pointThreshold: 0,
+      admissionProfile: undefined,
+      vestingInfomation: {
+        percentage: 0,
+        cliff: 0,
+        duration: 0,
+        slicePeriodSeconds: 0,
+      },
+    },
+    poolUnlimited: {
+      raisingAmountPool: BIG_ZERO,
+      offeringAmountPool: BIG_ZERO,
+      limitPerUserInLP: BIG_ZERO,
+      taxRate: 0,
+      totalAmountPool: BIG_ZERO,
+      sumTaxesOverflow: BIG_ZERO,
+      vestingInfomation: {
+        percentage: 0,
+        cliff: 0,
+        duration: 0,
+        slicePeriodSeconds: 0,
+      },
+    },
+    thresholdPoints: undefined,
+    startBlockNum: 0,
+    endBlockNum: 0,
+    numberPoints: 0,
+    vestingStartTime: 0,
+  })
+
+  const abi = version >= 3.1 ? ifoV3Abi : ifoV2Abi // ifoV2Abi use for version 3.0
+
+  const fetchIfoData = useCallback(
+    async (currentBlock: number) => {
+      const [
+        startBlock,
+        endBlock,
+        poolBasic,
+        poolUnlimited,
+        taxRate,
+        numberPoints,
+        thresholdPoints,
+        admissionProfile,
+        pointThreshold,
+        vestingStartTime,
+        basicVestingInformation,
+        unlimitedVestingInformation,
+      ] = await multicallv2(
+        abi,
+        [
+          {
+            address,
+            name: 'startBlock',
+          },
+          {
+            address,
+            name: 'endBlock',
+          },
+          {
+            address,
+            name: 'viewPoolInformation',
+            params: [0],
+          },
+          {
+            address,
+            name: 'viewPoolInformation',
+            params: [1],
+          },
+          {
+            address,
+            name: 'viewPoolTaxRateOverflow',
+            params: [1],
+          },
+          {
+            address,
+            name: 'numberPoints',
+          },
+          {
+            address,
+            name: 'thresholdPoints',
+          },
+          version >= 3.1 && {
+            address,
+            name: 'admissionProfile',
+          },
+          version >= 3.1 && {
+            address,
+            name: 'pointThreshold',
+          },
+          version === 3.2 && {
+            address,
+            name: 'vestingStartTime',
+          },
+          version === 3.2 && {
+            address,
+            name: 'viewPoolVestingInformation',
+            params: [0],
+          },
+          version === 3.2 && {
+            address,
+            name: 'viewPoolVestingInformation',
+            params: [1],
+          },
+        ].filter(Boolean),
+      )
+
+      const poolBasicFormatted = formatPool(poolBasic)
+      const poolUnlimitedFormatted = formatPool(poolUnlimited)
+
+      const startBlockNum = startBlock ? startBlock[0].toNumber() : 0
+      const endBlockNum = endBlock ? endBlock[0].toNumber() : 0
+      const taxRateNum = taxRate ? FixedNumber.from(taxRate[0]).divUnsafe(TAX_PRECISION).toUnsafeFloat() : 0
```

Comment:
i'm partially ok with using bignumberjs for new feature, but why mixing both bignumber.js and ethers?